scripts/build-ci.sh

```diff
@@ -0,0 +1,22 @@
+#!/bin/bash
+# Build script for continuous integration builds
+set -ex
+
+pushd `dirname $(readlink -f $0)/`/..
+
+if [ "$CIRCLE_BRANCH" == 'master' ]; then
+  original_version=`node -p "require('./package.json').version"`
+  # Set a nice build number when building master on CircleCI
+  # (eg. "0.16.0-alpha.123+20161018")
+  version="$original_version-alpha.$CIRCLE_BUILD_NUM+"`date +%Y%m%d`
+  YARN_VERSION_GIT_TAG='' yarn version --new-version $version
```

Comment:
Instead of $CIRCLE_BRANCH == 'master' you can use "branches" in deployment area https://github.com/facebook/react-native/blob/master/circle.yml#L97.

The nice thing about it is that it runs only when all Test phases are green.
Also does `yarn version` add a tag to git checkout?
I think it might not work becuase most likely CI checkouts have read-only access to repos.
That is why we store a special write token for RN tagging https://github.com/facebook/react-native/blob/master/circle.yml#L102